darts/models/torch_forecasting_model.py

```diff
@@ -571,20 +587,25 @@ def predict_from_dataset(self,
                 input_series = batch_tuple[0]
                 cov_future = batch_tuple[1] if len(batch_tuple) == 3 else None
 
-                if self.is_recurrent:
-                    batch_prediction = self._predict_batch_recurrent_model(n, input_series, cov_future)
-                else:
-                    batch_prediction = self._predict_batch_block_model(n, input_series, cov_future, roll_size)
+                # repeat prediction procedure for every needed sample
+                batch_predictions = []
+                for i in range(num_samples):
```

Comment:
Actually, isn't there a way to "tile" the `input_series` and `cov_future`, in a way that would allow only a simple call to `predict_*_model()` using batching? That would make things faster.